src/test/java/org/cactoos/iterator/NoNullsTest.java

```diff
@@ -23,72 +23,53 @@
  */
 package org.cactoos.iterator;
 
-import java.util.Iterator;
-import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.ExpectedException;
 import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.MatcherOf;
 import org.llorllale.cactoos.matchers.Throws;
 
 /**
  * Test cases for {@link NoNulls}.
  *
  * <p>There is no thread-safety guarantee.
- *
- * @since 0.35
  * @checkstyle JavadocMethodCheck (500 lines)
+ * @since 0.35
  */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
 public final class NoNullsTest {
 
-    /**
-     * A rule for handling an exception.
-     */
-    @Rule
-    public final ExpectedException exception = ExpectedException.none();
-
     @Test
     public void nextThrowsErrorIfNull() {
```

Comment:
@vzurauskas isn't this test just a clone of the second one? I don't see the distinction, they both test that a call to `next()` raises an exception. I suggest removing the first and renaming the second as `nextThrowsErrorIfNull`.

Reply:
@scristalli They are quite similar, but not equivalent. First one is the case where the first element is null, the second one is where the null is among non-nulls. It's a valid point that they might be too similar to warrant keeping them both, but I'm sure the original DEV had his reasons for having both tests as well. Since it was approved by both REV and ARC, I'm reluctant to remove one without good reason.

Reply:
@vzurauskas OK, let's keep both of them. Could you at least consider renaming? Right now I find the names a little confusing, I was thinking maybe `nextThrowsErrorIfFirstNull` and `nextThrowsErrorIfNthNull` or something similar, to highlight the difference.

Reply:
@scristalli I actually find the current names clearer. "Next throws error if null" means "first 'next()' throws error if there is a null". "Nth throws error if null" means "Nth 'next()' throws error if there is a null." Again, I'd rather not change something already approved by a REV and an ARC without good reason.